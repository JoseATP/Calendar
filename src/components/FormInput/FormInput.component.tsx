import { FC, InputHTMLAttributes } from "react";
import { FormInputLabel, Group, Input } from "./FormInput.styles";

type FormInputProps = {
  label?: string;
} & InputHTMLAttributes<HTMLInputElement>;

const FormInput: FC<FormInputProps> = ({ label, ...otherProps }) => {
  return (
    <Group>
      {label && <FormInputLabel>{label}</FormInputLabel>}
      <Input {...otherProps} />
    </Group>
  );
};

export default FormInput;
