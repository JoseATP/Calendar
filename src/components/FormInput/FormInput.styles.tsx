import styled from "styled-components";
import { colors } from "../../constants/colors.constants";

export const FormInputLabel = styled.label`
  display: block;
  color: ${colors.blackLight};
  font-size: 16px;
  font-weight: normal;
  pointer-events: none;
`;


export const Input = styled.input`
  background: none;
  background-color: white;
  color: ${colors.blackLight};
  font-size: 18px;
  padding: 10px 10px 10px 5px;
  display: block;
  width: 100%;
  border-radius: 5px;
  border: 1px solid rgba(49, 39, 37, 0.4);
  margin: 5px 0px;
  ::placeholder {
    color: rgba(56, 64, 66, 0.5);
    font-size: 1rem;
  }
  :-ms-input-placeholder { /* Internet Explorer 10-11 */
    color: rgba(56, 64, 66, 0.5);
    font-size: 1rem ;
  }

  ::-ms-input-placeholder { /* Microsoft Edge */
    color: rgba(56, 64, 66, 0.5);
    font-size: 1rem;
  }

`;



export const Group = styled.div`
  position: relative;
  width: 100%;
  margin: 30px 0;

  input[type="password"] {
    letter-spacing: 0.3em;
  }
`;
