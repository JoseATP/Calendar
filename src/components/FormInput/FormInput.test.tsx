import { render, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import FormInput from "./FormInput.component";

afterEach(cleanup);

describe("Form Input", () => {
  it("Should render FormInput component correctly", () => {
    const { asFragment } = render(<FormInput label="test" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Check formInput type", async () => {
    render(<FormInput />);

    const FormInputElem = screen.getByRole("textbox");

    userEvent.clear(FormInputElem);
    userEvent.type(FormInputElem, "Title");

    expect(FormInputElem).toHaveValue("Title");
  });
});
