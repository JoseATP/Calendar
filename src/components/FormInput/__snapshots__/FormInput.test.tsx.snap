// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Input Should render FormInput component correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-dmyDGi VDlCT"
  >
    <label
      class="sc-beySbM dgdslS"
    >
      test
    </label>
    <input
      class="sc-guDLRT hoErlo"
    />
  </div>
</DocumentFragment>
`;
